pkg/template/template.go

```diff
@@ -41,16 +41,17 @@ type (
 	// ConfigureTypeFn dispatches the inferred types to handlers
 	ConfigureTypeFn func(types map[string]proto.Message, builder *adapter.HandlerBuilder) error
 
-	// ProcessReportFn instantiates the instance object and dispatches them to the handler.
-	ProcessReportFn func(allCnstrs map[string]proto.Message, attrs attribute.Bag, mapper expr.Evaluator, handler adapter.Handler) rpc.Status
+	// EvaluateFn creates the instance object by evaluating the constructor config using the attributes
+	EvaluateFn func(instName string, cnstrParam proto.Message, attrs attribute.Bag, mapper expr.Evaluator) (interface{}, error)
 
-	// ProcessCheckFn instantiates the instance object and dispatches them to the handler.
-	ProcessCheckFn func(instName string, cnstr proto.Message, attrs attribute.Bag, mapper expr.Evaluator,
-		handler adapter.Handler) (rpc.Status, adapter.CacheabilityInfo)
+	// DispatchReportFn dispatches the instance to the handler.
+	DispatchReportFn func(context.Context, interface{}, adapter.Handler) (adapter.ReportResult, error)
 
-	// ProcessQuotaFn instantiates the instance object and dispatches them to the handler.
-	ProcessQuotaFn func(quotaName string, cnstr proto.Message, attrs attribute.Bag, mapper expr.Evaluator, handler adapter.Handler,
-		args adapter.QuotaRequestArgs) (rpc.Status, adapter.CacheabilityInfo, adapter.QuotaResult)
+	// DispatchCheckFn dispatches the instance to the handler.
+	DispatchCheckFn func(context.Context, interface{}, adapter.Handler) (adapter.CheckResult, error)
```

Comment:
old model's sig for  quota had adapter.QuotaResult, so, I am trying to follow the same convention. But I suppose we can make this pointer. @geeknoid any concerns ?